### PR TITLE
[NEW][Apps] Add a new upload API

### PR DESCRIPTION
**Pull request by lolimay**
## Proposed changes

Add a new upload API `createUpload` under the upload bridge

## Issue(s)

https://github.com/RocketChat/Rocket.Chat.Apps-engine/pull/318

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Hotfix (a major bugfix that has to be merged asap)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat)
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [x] I have added necessary documentation (if applicable)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by lolimay** [comment]
Depends on https://github.com/RocketChat/Rocket.Chat.Apps-engine/pull/318

File: package.json [hunk 1]
```diff
@@ -163,6 +163,7 @@
 		"cas": "https://github.com/kcbanner/node-cas/tarball/fcd27dad333223b3b75a048bce27973fb3ca0f62",
 		"change-case": "^4.1.1",
 		"chart.js": "^2.9.3",
+		"clean": "^4.0.2",
```
Comment (d-gubert) [inline package.json:166]:
```suggestion
```

**Review by d-gubert — DISMISSED** [review]

**Review by d-gubert — APPROVED** [review]

File: app/file-upload/server/lib/FileUpload.js [hunk 2]
```diff
@@ -66,7 +66,7 @@ export const FileUpload = {
 		const room = Rooms.findOneById(file.rid);
 		const directMessageAllowed = settings.get('FileUpload_Enabled_Direct');
 		const fileUploadAllowed = settings.get('FileUpload_Enabled');
-		if (canAccessRoom(room, user, file) !== true) {
+		if (canAccessRoom(room, user, file) !== true && user.type !== 'app') {
```
Comment (sampaiodiego) [inline app/file-upload/server/lib/FileUpload.js:69]: `canAccessRoom` is a heavy operation, if app users will bypass it, we can bypass calling it:
```suggestion
		if (user.type !== 'app' && canAccessRoom(room, user, file) !== true) {
```

**Review by sampaiodiego — CHANGES_REQUESTED** [review]

**Comment by d-gubert** [comment]
@lolimay @thassiov @sampaiodiego currently, this does **not** send a message to the specified room after the file is uploaded. Should we do this? Or should we leave it as a responsibility for the app to send it?

**Comment by sampaiodiego** [comment]
> @lolimay @thassiov @sampaiodiego currently, this does **not** send a message to the specified room after the file is uploaded. Should we do this? Or should we leave it as a responsibility for the app to send it?

in rocket.chat there is no API to send a file to a room without sending a message, IMHO this behavior should be applied to the engine as well

**Review by sampaiodiego — APPROVED** [review]

**Pull request merged**